packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmUUIDImpl.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.internal
+
+import io.realm.kotlin.internal.interop.UUIDWrapper
+import io.realm.kotlin.internal.util.HEX_PATTERN
+import io.realm.kotlin.internal.util.parseHex
+import io.realm.kotlin.internal.util.toHexString
+import io.realm.kotlin.types.RealmUUID
+import kotlin.experimental.and
+import kotlin.experimental.or
+import kotlin.random.Random
+
+@Suppress("MagicNumber")
+internal class RealmUUIDImpl : RealmUUID, UUIDWrapper {
+
+    private val _bytes: ByteArray
+
+    override val bytes: ByteArray
+        get() = _bytes
+
+    constructor() {
+        _bytes = Random.nextBytes(UUID_BYTE_SIZE).apply {
+            // Set uuid to version 4, 6th byte must be 0x40
```

Comment:
Is this right? Or is the comment just wrong? Maybe it should just be
```suggestion
            // Set uuid to version 4, 6th byte must be 0x4x
```